client/js/socket-events/connection.js

```diff
@@ -39,6 +42,18 @@ function handleDisconnect(data) {
 		return;
 	}
 
+	// Sorry brunnre8, but I have to consume the error message since there appears to be no way to
+	// get the error from anywhere else
+	// We are checking if the server configuration is null because if it is null, then the client
+	// never loaded properly and there's a different issue than 401 on the header auth
+	if (
+		store.state.serverConfiguration !== null &&
+		message === "xhr poll error" &&
```

Comment:
This is still not okay, there has to be a better way.

Run TL with the following config:

```json
// ### `transports`
	//
	// Set `socket.io` transports.
	//
	// This value is set to `["polling", "websocket"]` by default.
	transports: ["websocket"],
```

This will break the above as far as I can tell.
Every single update of socket.io may break that and we'd have nothing to complain about.
Error messages are not part of the API.

Reply:
It seems like error messages are used for handing disconnect errors already: https://github.com/thelounge/thelounge/blob/master/client/js/socket-events/connection.js#L48

Reply:
I had the same thought as @brunnre8 and then the same follow-up thought as @9p4. You should make sure there's no error code on the object that we can use instead, but checking the error message is sometimes the only option 🤷‍♀️ .